src/components/app/app.js

```diff
@@ -68,20 +78,58 @@ export default class App extends Component {
         notes: notes,
       };
     });
+
+    fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
+      method: "PUT",
+      body: JSON.stringify(this.state.notes),
+    });
+  };
+
+  searchNotes(notesMap, text) {
+    if (text.length === 0) {
+      return Array.from(notesMap).map(([, note]) => note);
+    }
+    const notes = [];
+    for (let keyValue of notesMap) {
+      const note = keyValue[1];
+      if (note.title.startsWith(text)) {
+        notes.push(note);
+      }
+    }
+    return notes;
+  }
+
+  listItems(data) {
+    let items = [];
+
+    Array.from(data).map((id) => items.push(id));
```

Comment:
I don't think that map function is needed, `Array.from` returns array from iterable collection
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from